src/library/scala/collection/IterableOnce.scala

```diff
@@ -977,16 +977,13 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
     *  @return      the string builder `b` to which elements were appended.
     */
   def addString(b: StringBuilder, start: String, sep: String, end: String): b.type = {
-    var first = true
     b.append(start)
-    for (x <- this) {
-      if (first) {
-        b.append(x)
-        first = false
-      }
-      else {
+    val it = iterator
+    if (it.hasNext) {
+      b.append(it.next)
+      while (it.hasNext) {
         b.append(sep)
-        b.append(x)
+        b.append(it.next)
```

Comment:
Would it be old-fashioned to `b.append(sep).append(it.next)`?

Reply:
that doesn't work because this is a scala stringbuilder. ideally we'd switch to java stringbuilder but it's a semi-public api.

Reply:
It works for me, but it's the end of a long day for me, you must be saying something I don't understand.

Reply:
yes, my mistake, you are correct.  I don't think it simplifies the readibility, though.